src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ClassConstructorContextNode.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a node with a type's class constructor context containing a pointer to the class
+    /// constructor method entrypoint and a flag field the runtime checks to ensure it only runs once.
+    /// </summary>
+    internal class ClassConstructorContextNode : ObjectNode, ISymbolNode
+    {
+        private MetadataType _type;
+        
+        public ClassConstructorContextNode(MetadataType type, NodeFactory factory)
+        {
+            _type = type;
+        }
+
+        public override string GetName()
+        {
+            return ((ISymbolNode)this).MangledName;
+        }
+
+        public override ObjectNodeSection Section
+        {
+            get
+            {
+                return ObjectNodeSection.DataSection;
+            }
+        }
+
+        public override bool ShouldShareNodeAcrossModules(NodeFactory factory)
+        {
+            return factory.CompilationModuleGroup.ShouldShareAcrossModules(_type);
+        }
+
+        string ISymbolNode.MangledName
+        {
+            get
+            {
+                return "__CCtorContext_" + NodeFactory.NameMangler.GetMangledTypeName(_type);
+            }
+        }
+
+        int ISymbolNode.Offset
+        {
+            get
+            {
+                return 0;
+            }
+        }
+
+        private int ClassConstructorContextSize
+        {
+            get
+            {
+                // TODO: Assert that StaticClassConstructionContext type has the expected size
+                //       (need to make it a well known type?)
+                return _type.Context.Target.PointerSize * 2;
+            }
+        }
+
+        private int ClassConstructorContextAlignment
```

Comment:
Nit: This is now used from exactly one place. Feels overengineered to have it as property.